pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -204,9 +205,24 @@ func makeVirtualServiceRoute(hosts sets.String, http *v1alpha1.HTTPIngressPath,
 		}
 	}
 
+	var rewrite *istiov1alpha3.HTTPRewrite
+	if http.RewriteHost != "" {
```

Comment:
It would be helpful to keep the original host header in case the application may want to consume it. WDYT?

Reply:
since this is relatively low-level (the plan is to build a more user-facing DomainMapping CRD or similar on top of it this feature rather than exposing KIngress to users) I was imagining any behaviour like that would happen higher up (i.e. by adding AppendHeaders to the spec). That way the higher-up code can decide if it wants to add such a header and what to call it. Does that sounds ok or do you think it makes more sense to make it an implicit part of the low-level RewriteHost flow?

Reply:
That sounds good to me.